src/table/components/TableBodyWrapper.jsx

```diff
@@ -75,72 +77,85 @@ function TableBodyWrapper({
     addSelectionListeners({ api: selectionsAPI, selDispatch, setShouldRefocus, keyboard, tableWrapperRef });
   }, []);
 
+  // useEffect(() => {
+  //   setSrNotation(
+  //     getCellSrNotation({
+  //       focusedCellCoord,
+  //       rootElement,
+  //       selectionState,
+  //       translator,
+  //       isActiveElementInTable,
+  //     })
+  //   );
+  // }, [focusedCellCoord, selectionState, translator, isActiveElementInTable]);
   useEffect(() => {
-    setSrNotation(
-      getCellSrNotation({
-        focusedCellCoord,
+    if (focusedCellCoord[0] === 0) return;
+
+    const announcementType = selectionsAPI.isModal()
+      ? ANNOUNCEMENT_TYPES.SELECTION_TYPE
+      : ANNOUNCEMENT_TYPES.FOCUS_TYPE;
+
+    emitAnnouncement({
+      announcementType,
+      notationDependencies: {
         rootElement,
         selectionState,
-        translator,
+        focusedCellCoord,
         isActiveElementInTable,
-      })
-    );
-  }, [focusedCellCoord, selectionState, translator, isActiveElementInTable]);
+        translator,
+      },
+    });
+  }, [focusedCellCoord, selectionState, selectionsAPI, isActiveElementInTable, translator]);
```

Comment:
So my idea of creating some kind of global aria-live component was that we want to get rid of this. Instead of listening on what has changed, we tell the Announcer exactly what it should say. This should be possible if we can call some kind of set function **from where we actually make a changes to the chart**. For example, in `selection-utils` there is a function updating the amount of rows. There is a case for removing and one for adding. So we already know what the screen reader should say. That means we can get rid of a lot of the logic in ´getMemoisedAnnouncementNotation´ and other places as well

Reply:
sometimes we dont have things like translator and some other required objects like layout or tableData, and other stuff in those functions, how we can pass those required objects , for example to determine which column is sorted for example. 

I think sometimes we need some external data objects to compare and get some results out of them, i'm thinking we could end up passing more arguments to functions like `updateFocus`, `selectCell` and others.